CMakeLists.txt

```diff
@@ -137,6 +137,16 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
     endif()
 endif()
 
+# Optional codegen target
+if(PYTHONINTERP_FOUND)
+    add_custom_target(VulkanTools_generated_source
+                      COMMAND ${PYTHON_EXECUTABLE} ${PROJECT_SOURCE_DIR}/scripts/generate_source.py
+                              ${VulkanRegistry_DIR} --incremental
+                      )
+else()
+    message("WARNING: VulkanTools_generated_source target requires python 3")
```

Comment:
I'm not sure `WARNING` is strong enough.  The `VulkanTools_generated_source` target should fail if Python3 is not available, yes?

Reply:
You just wouldn't get a `VulkanTools_generated_source` target if you don't have Python3.  This is similar to the way LunarG/VulkanTools handles the optional Qt5 dependency for the vkconfig target: https://github.com/LunarG/VulkanTools/blob/8792090d531807522c808663b10043da4a2f502c/vkconfig/CMakeLists.txt#L12